src/loaders.ts

```diff
@@ -2,6 +2,17 @@ import { addressWithChecksum, fetchJSON } from "./utils.js";
 
 export interface ABILoader {
   loadABI(address: string): Promise<any[]>;
+  getContract(address: string): Promise<ContractData | null>
+}
+
+type EvmVersion = 'homestead' | 'tangerine whistle' | 'spurious dragon' | 'byzantium' | 'constantinople' | 'istanbul' | 'berlin' | 'london' | 'paris' | 'shanghai'
```

Comment:
Think it's worth maintaining this enum versus just letting it be a plain string for now? (Is this something that would make things easier for you?)

Reply:
I'll drop it for a plain string. There might be differences in casing, etc. between etherscan and sourcify that I haven't investigated yet.

Reply:
Sounds like an easier starting point, thanks! Ultimately I'd like to introduce more concrete types to help the devex, but I haven't decided what the best approach is while still staying compatible with as much EVM-based chains as possible and without having a lot of update churn to keep up with changes.

Was thinking of maybe using a third party library like abitype or maybe a component of ethers.js.

Reply:
I think using abiType would be a good addition.